pkg/proxy/nftables/proxier.go

```diff
@@ -320,7 +320,7 @@ func getNFTablesInterface(ipFamily v1.IPFamily) (knftables.Interface, error) {
 	//
 	// However, we allow the user to bypass this check by setting
 	// `KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK` to anything non-empty.
-	if os.Getenv("KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK") != "" {
+	if os.Getenv("KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK") == "" {
```

Comment:
we need to check if this matches @thockin expectaction in https://github.com/kubernetes/kubernetes/pull/124152#discussion_r1666105310

Reply:
reading this comments I think he is advocating for https://pkg.go.dev/os#LookupEnv

Reply:
```suggestion
	if _, ok := os.Getenv("KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK") ; !ok {
```

Reply:
In this case, if we use `os.LookupEnv` with checking ok only instead of `os.Getenv`, we cannot skip the check when passing empty string. In order to skip the check if the env is set but empty string, we need to check the value itself.

---

```
import (
        "os"
        "fmt"
)

func main() {
        fmt.Println(os.Getenv("KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK"))
        fmt.Println(os.LookupEnv("KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK"))
}
```

```
% go run main.go

 false

% KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK= go run main.go

 true

% KUBE_PROXY_NFTABLES_SKIP_KERNEL_VERSION_CHECK=skip go run main.go
skip
skip true
```

Reply:
I got it wrong, sorry